### PR TITLE
Ignore modifiers in WASD plugin input

### LINKED ISSUE #7165: Allow WASD Camera movement while shift is enabled
The WASD feature is amazing, but when using sticky keys to make it easier to drop items/just having it be more useful to have enabled in general, I am unable to use the awesome WASD camera movement and have to rely on the mouse-wheel instead

I have no idea how coding works but this would be very useful as I wouldn't have to keep turning on and off sticky keys which is somewhat rough considering I can't remap keys on my keyboard.

So yeah, being able to move camera with WASD while shift is enabled, if anyone can make this happen I'd be forever grateful

### PR DESCRIPTION
**Pull request by Abextm**
Supersedes/Closes #7369
Supersedes/Closes #7331
Closes #7165

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/config/Keybind.java [hunk 1]
```diff
@@ -60,7 +60,7 @@
 	private final int keyCode;
 	private final int modifiers;
 
-	public Keybind(int keyCode, int modifiers)
+	protected Keybind(int keyCode, int modifiers, boolean ignoreModifiers)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/config/Keybind.java]: maybe this constructor should be just keyCode, ignoreModifiers? as passing modifiers when you want to ignore them do not makes much sense. Or even just keyCode, even though that can maybe be a bit confusing idk

File: runelite-client/src/main/java/net/runelite/client/config/ConfigManager.java [hunk 2]
```diff
@@ -514,11 +514,15 @@ static Object stringToObject(String str, Class<?> type)
...
 		}
-		if (type == Keybind.class)
+		if (type == Keybind.class || type == ModifierlessKeybind.class)
 		{
 			String[] splitStr = str.split(":");
 			int code = Integer.parseInt(splitStr[0]);
 			int mods = Integer.parseInt(splitStr[1]);
+			if (type == ModifierlessKeybind.class)
+			{
+				return new ModifierlessKeybind(code, mods);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/config/ConfigManager.java]: Again here, storing mods and then using them for modifierLessKeybind do not makes much sense

Comment (Abextm) [reply runelite-client/src/main/java/net/runelite/client/config/ConfigManager.java]: same thing as above

File: runelite-client/src/main/java/net/runelite/client/config/Keybind.java (hunk 1 above)
Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/config/Keybind.java]: But shift is also KeyCode i believe, and so you dont need to do that

Comment (Abextm) [reply runelite-client/src/main/java/net/runelite/client/config/Keybind.java]: internally (and therefore through this constructor too) modifier only keys are stored as `VK_UNDEFINED` with modifiers set

**Pull request merged**